app/assets/javascripts/discourse/app/components/composer-editor.gjs

```diff
@@ -130,17 +131,24 @@ export default class ComposerEditor extends Component {
 
   @discourseComputed("composer.model.requiredCategoryMissing")
   replyPlaceholder(requiredCategoryMissing) {
+    let placeholder;
     if (requiredCategoryMissing) {
-      return "composer.reply_placeholder_choose_category";
+      placeholder = "composer.reply_placeholder_choose_category";
     } else {
       const key = authorizesOneOrMoreImageExtensions(
         this.currentUser.staff,
         this.siteSettings
       )
         ? "reply_placeholder"
         : "reply_placeholder_no_images";
-      return `composer.${key}`;
+      placeholder = `composer.${key}`;
     }
+
+    return applyValueTransformer(
+      "composer-editor-reply-placeholder",
+      placeholder,
+      { model: this.composer }
+    );
```

Comment:
It's fine, a bit of 🏌️ 

Maybe something like this seems better?

```js
  let placeholder = "composer.reply_placeholder_choose_category";

  if (!requiredCategoryMissing) {
    const key = authorizesOneOrMoreImageExtensions(
      this.currentUser.staff,
      this.siteSettings
    )
      ? "reply_placeholder"
      : "reply_placeholder_no_images";
    placeholder = `composer.${key}`;
  }

  return applyValueTransformer(
    "composer-editor-reply-placeholder",
    placeholder,
    { model: this.composer }
  );
```

Reply:
good call -- I initially didn't like this b/c it makes the "base case" something that is `if`'d but whatever, this is better. will commit.